File: ACTIV.py
# will get TLE

import bisect as bs
mod = int(1e8)

def solve():
	dp = [[0, 0] for i in range(n)]
	ls = list()
	dp[0][1] = 1
	ls.append(arr[0][0])
	for i in range(1, n):
		dp[i][0] = (dp[i-1][0] + dp[i-1][1]) % mod
		if arr[i][1] < ls[0]:
			dp[i][1] = 1
		else:
			pos = bs.bisect_right(ls, arr[i][1])
			pos -= 1
			dp[i][1] = (1 + dp[pos][0] + dp[pos][1]) % mod
		ls.append(arr[i][0])
	res = (dp[n-1][0] + dp[n-1][1]) % mod
	ans = ('0' * (8 - len(str(res)))) + str(res)
	print(ans)

while True:
	n = int(input())
	if n == -1:
		break
	arr = [list(map(int, input().split())) for i in range(n)]
	for i in range(n):
		arr[i][0], arr[i][1] = arr[i][1], arr[i][0]
	arr.sort()
	solve()	

File: ACTIV.cpp
#include <bits/stdc++.h>
using namespace std;
#define endl '\n'

long mod = 1e8;
int n;
vector<pair<int, int>> arr;

void solve() {
	int dp[n][2];
	vector<int> v;
	memset(dp, 0, sizeof(dp));
	dp[0][1] = 1;
	v.push_back(arr[0].first);
	for (int i = 1; i < n; i++) {
		dp[i][0] = (dp[i-1][0] + dp[i-1][1]) % mod;
		if (arr[i].second < v[0]) {
			dp[i][1] = 1;
		} else {
			int pos = upper_bound(v.begin(), v.end(), arr[i].second) - v.begin();
			pos--;
			dp[i][1] = (1 + dp[pos][0] + dp[pos][1]) % mod;
		}
		v.push_back(arr[i].first);
	}
	printf("%0.8d\n", (dp[n-1][0] + dp[n-1][1]) % mod);
}

int main(void) {
	ios::sync_with_stdio(false);
	cin.tie(0);
	cout.tie(0);

#ifndef ONLINE_JUDGE
	freopen("input.txt", "r", stdin);
//	freopen("output.txt", "w", stdout);
#endif

	while (cin >> n) {
		if (n == -1)
			break;
		arr.resize(n);
		for (int i = 0; i < n; i++)
			cin >> arr[i].second >> arr[i].first;
		sort(arr.begin(), arr.end());
		solve();
	}
}
